dev_tools/nyaml2nxdl/nyaml2nxdl_backward_tools.py

```diff
@@ -0,0 +1,1046 @@
+#!/usr/bin/env python3
+"""This file collects the function used in the reverse tool nxdl2yaml.
+
+"""
+import os
+
+# -*- coding: utf-8 -*-
+#
+# Copyright The NOMAD Authors.
+#
+# This file is part of NOMAD. See https://nomad-lab.eu for further info.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import sys
+import xml.etree.ElementTree as ET
+from typing import Dict
+from typing import List
+
+from .nyaml2nxdl_helper import cleaning_empty_lines
+from .nyaml2nxdl_helper import get_node_parent_info
+from .nyaml2nxdl_helper import get_yaml_escape_char_dict
+from .nyaml2nxdl_helper import remove_namespace_from_tag
+
+DEPTH_SIZE = "  "
+CMNT_TAG = "!--"
+
+
+def separate_pi_comments(input_file):
+    """
+    Separate PI comments from ProcessesInstruction (pi)
+    """
+    comments_list = []
+    comment = []
+    xml_lines = []
+
+    with open(input_file, "r", encoding="utf-8") as file:
+        lines = file.readlines()
+        has_pi = True
+        for line in lines:
+            c_start = "<!--"
+            cmnt_end = "-->"
+            def_tag = "<definition"
```

Comment:
It would not add any significant difference if they are defined at the module level. As these are only utilized inside that function, so unnecessary at the module level.

Reply:
:+1: